### DISCUSSION TITLE
Search for folders and file names

### DISCUSSION THREAD

**Discussion by R-W-C** [hvianna/audioMotion.js#56] [Ideas]
Hi,

First of all, let me tell me how impressed I am with your work. 

There seems to be once important feature missing, the possibility to search for folder and file names. Or is it somewhere hidden?

----

And another thing, but really not that important. The number of channels is not showed in the information and neither is the bitrate. 
Mono channel files and multichannel flac files are played just fine, so it makes sense to display the number of channels. Or is it something I have to switch on?

**Comment by JesusGod-Pope666-Info** [comment]
![Screenshot_2024-04-11_15-48-16](https://github.com/hvianna/audioMotion.js/assets/90806094/7ef97175-8ab1-444a-b7f8-dccd5a6ecbd6)
The information of the file in regards of bitrate is at the right side in the visualizer window. Here a very low mp3 file, although from a Radio show. I think it is mono, but it might not display that.... not sure, and less of a concern really. But.... sure if there is a an easy way to display it that looks good. 1C or 2C.... or however it is normally done, Mono, Stereo..... Dunno. I guess it could be looked into the 80 for if they had some kind of feature for Mono or Stereo.... But I don't know.

**Reply by R-W-C** [reply]
Well, I would show it like Foobar 2000 does. 

Bitrate: 320 kbps
Channels: 5.1 

Both bitrate and number of channels is present in all the audio files, so that should be a piece of cake.
I don't care about mono or stereo, because mono can be played over 2 channels (which doesn't make it stereo of course).
Why is it important? Because I have got different versions of some songs, records and it would be nice to see what version I am dealing with.  

How about being able to search?

**Reply by JesusGod-Pope666-Info** [reply]
Well bitrate is already shown, and file type as shown on the image with the data at the right side. I guess channels could be added as well after the bitrate data not taking up to much space, although for myself, not sure how useful it would be.

But again you already have the file type and Bitrate shown, and if Channels is added, you would have every data you ask for.

**Comment by hvianna** [comment]
I currently prioritize displaying the sampling rate and bit depth (e.g. 96kHz / 24bit) when these are available in the metadata, because for lossless and high-definition formats I think these are more useful information than the bitrate. But I can try fitting everything together, with the additional bitrate and channel count.

The search feature is a good idea, but it requires rethinking how the file explorer works. I opted for an old-school file navigation instead of the modern "media library" approach, where the software takes hours scanning your entire hard drive and caches the content (requiring an occasional "refresh" whenever files are updated). A search feature would probably require something that, which I don't really like.

Anyways, I'll create issues for both of these requests so I keep them in mind.

https://github.com/hvianna/audioMotion.js/issues/57
https://github.com/hvianna/audioMotion.js/issues/58
